Packs/MandiantAdvantageAttackSurfaceManagement/Integrations/AttackSurfaceManagement/AttackSurfaceManagement.py

```diff
@@ -0,0 +1,520 @@
+        if since:
+            datetime_query = datetime_query + f' last_seen_after:{since.strftime("%Y-%m-%dT%H:%M:%S.%fZ")}'
+
+        search_query = collections_query + datetime_query
+        endpoint = endpoint + urllib.parse.quote(search_query)
+
+        params = {'page': 0}
+
+        while True:
+            response = self.make_request('GET', endpoint, params=params,
+                                         headers={'PROJECT_ID': str(self.project_id)})
+
+            if not isinstance(response, dict) or not response.get('success'):
+                raise RuntimeError('Failed to retrieve issues from ASM')
+
+            yield from response.get('result', {}).get('hits')
+
+            if response.get('result', {}).get('more'):
+                params['page'] = params['page'] + 1
+            else:
+                break
+
+    def get_issue_details(self,
+                          issue_id: str) -> dict:
+        endpoint = f'issues/{issue_id}'
+
+        response = self.make_request('GET', endpoint,
+                                     headers={'PROJECT_ID': str(self.project_id)})
+
+        if not isinstance(response, dict) or not response.get('success'):
+            raise DemistoException('The ASM API was unable to return details for'
+                                   f'issue {issue_id} in project {self.project_id}')
+
+        return response['result']
+
+    def get_notes(self,
+                  resource_type: str,
+                  resource_id: str) -> dict:
+        endpoint = f'notes/{resource_type}/{resource_id}'
+
+        response = self.make_request('GET', endpoint,
+                                     headers={'PROJECT_ID': str(self.project_id)})
+
+        if not isinstance(response, dict) or not response.get('success'):
+            raise DemistoException('The ASM API was unable to return notes for'
+                                   f'{resource_type} {resource_id} in project {self.project_id}')
+
+        return response['result']
+
+    # TODO: REMOVE the following dummy function:
+    def baseintegration_dummy(self, dummy: str) -> Dict[str, str]:
+        """Returns a simple python dict with the information provided
+        in the input (dummy).
+
+        :type dummy: ``str``
+        :param dummy: string to add in the dummy dict that is returned
+
+        :return: dict as {"dummy": dummy}
+        :rtype: ``str``
+        """
+
+        return {"dummy": dummy}
+    # TODO: ADD HERE THE FUNCTIONS TO INTERACT WITH YOUR PRODUCT API
+
+
+''' HELPER FUNCTIONS '''
+
+# TODO: ADD HERE ANY HELPER FUNCTION YOU MIGHT NEED (if any)
+
+
+def helper_create_incident(issue: dict, project_id: int) -> dict:
+    severity_map = {1: 4, 2: 3, 3: 2, 4: 1, 5: 0.5}
+
+    raw_json = issue
+    raw_json['mirror_direction'] = 'Both'
+    raw_json['mirror_instance'] = demisto.integrationInstance()
+
+    issue = {
+        'name': issue['summary']['pretty_name'],
```

Comment:
```suggestion
        'name': issue.get('summary', {}).get('pretty_name'),
```

when handling `dict` please use the `get` method of `dict` class to avoid situations where you have `KeyError`.

please change it for all the rest too.

Reply:
@chrishultin does not seem like it has been implemented